File: spring-integration/routing-messages/payload-type-router/src/main/java/com/ddubson/integration/services/DefaultPrintService.java
package com.ddubson.integration.services;

import org.springframework.messaging.Message;
import org.springframework.stereotype.Component;

@Component
public class DefaultPrintService implements PrintService {
    @Override
    public void print(Message<String> msg) {
        System.out.format("[%s] Default Channel: %s\n", Thread.currentThread().getName(), msg.getPayload());
    }
}

File: spring-integration/routing-messages/payload-type-router/src/main/kotlin/com/ddubson/integration/services/DefaultPrintService.kt
package com.ddubson.integration.services

import org.springframework.messaging.Message
import org.springframework.stereotype.Component

@Component
class DefaultPrintService : PrintService {
    override fun print(msg: Message<*>) {
        println("[${Thread.currentThread().name}] Default service: ${msg.payload}")
    }
}